src/xmlnode.cpp

```diff
@@ -221,12 +217,8 @@ std::string XmlNode::name() const
 
 bool XmlNode::hasAttribute(const char *attributeName) const
 {
-    bool found = false;
     xmlAttrPtr attribute = xmlHasProp(mPimpl->mXmlNodePtr, reinterpret_cast<const xmlChar *>(attributeName));
-    if (attribute != nullptr) {
-        found = true;
-    }
-    return found;
+    return (attribute == nullptr) ? false : true;
```

Comment:
Might just want to have `return attribute != nullptr;`.

Reply:
[b289591](https://github.com/cellml/libcellml/pull/1113/commits/b2895912c85642a887048534320cf5489d1553f6)